File: NN_DASH/nn_dash/app/src/main/java/com/example/nn_dashboard/ProductDao.java
package com.example.nn_dashboard;


import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface ProductDao {
    @Query("SELECT * FROM tbl_products")
    List<Product> getAllProducts();

    @Insert
    void insertProduct(Product product);
}

File: DASH-ROOM/test1_dash_room/app/src/main/java/com/example/test1_dash_room/ProductDao.kt
package com.example.test1_dash_room.data.local

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.example.test1_dash_room.Product

@Dao
interface ProductDao {
    @Query("SELECT * FROM tbl_products")
    suspend fun getAllProducts(): List<Product>  // Changed return type to List<Product>

    @Insert
    suspend fun insertProduct(product: Product)
}
